### PR TITLE
69 basic mapping and drive foward

### LINKED ISSUE #69: Basic mapping and drive foward
As a driver, I want the car to be able to drive forward using an internal map of the parking lot, so that I don't have to find a parking spot and do the driving myself

acceptance criteria:
- [x] The car needs an internal map that mapps the parking lot
- [x] The car should be able to assign itself to a empty parking spot using the map
- [x] The car should be able to drive towards the assigned parking spot

### PR DESCRIPTION
**Pull request by sh4r10**
The car has an idea of where it is relative to the parking lot and can now drive up to an unoccupied parking space.  This closes #69.

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

**Comment by sh4r10** [comment]
The car needs to assign the chosen parking spot as occupied or something else to that effect .

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -55,6 +55,71 @@ const int BACKWARD_SPEED_LIMIT = -50;
...
+    {GridBox(Occupied), GridBox(Path), GridBox(Occupied)},
+    //10                11             12
+    {GridBox(Unoccupied), GridBox(Path), GridBox(Occupied)},
+    //20                21             22
+    {GridBox(Occupied), GridBox(Path), GridBox(Occupied)},
+    //30                31             32
+    {GridBox(Occupied), GridBox(Path), GridBox(Occupied)},
+    //40                41             42
+    {GridBox(NaP),      GridBox(Start),GridBox(NaP)},
+};
```
Comment (arrestoteles) [inline arduino/smartcar/smartcar.ino:89]: Comment would be good to add.

Also, would it be easy if we would like to scale our parking lot? And how would it look like if the parking lot was not rectangular?

Besides that, awesome work!

Comment (sh4r10) [reply arduino/smartcar/smartcar.ino:93]: Thanks for the suggestion, I have made the appropriate changes. Please let me know if the changes live up to your expectations.

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -55,6 +55,71 @@ const int BACKWARD_SPEED_LIMIT = -50;
...
+    for(int i = 0; i < PARKING_ROWS; i++){
+        for(int j = 0; j<PARKING_COLS; j++){
+            if(parkingLot[i][j].type == Unoccupied){
+                move(ENTRANCE_R, ENTRANCE_C, i, j);
+            }
+        }
+    }
+}
+
+void move(int r1, int c1, int r2, int c2){
```
[the remark this answers is not in the document]
Comment (Druner) [reply arduino/smartcar/smartcar.ino:106]: Correct

File: arduino/smartcar/smartcar.ino [hunk 3]
```diff
@@ -55,6 +55,71 @@ const int BACKWARD_SPEED_LIMIT = -50;
 const int MAX_STEERING_ANGLE = 60;
 const auto ONE_SECOND = 1000UL;
 
+const int BOX_WIDTH = 70;
+const int BOX_HEIGHT = 35;
+const int PARKING_ROWS = 5;
+const int PARKING_COLS = 3;
+const int ENTRANCE_R = 4;
```
[the remark this answers is not in the document]
Comment (sh4r10) [reply arduino/smartcar/smartcar.ino:62]: Entrance R is the Row in terms of the map where the car starts, and Entrance C is the Column of where the car can start to park.

File: arduino/smartcar/smartcar.ino (hunk 2 above)
[the remark this answers is not in the document]
Comment (sh4r10) [reply arduino/smartcar/smartcar.ino:106]: yes r1, c1 is the entrance box and the r2, c2 is the intendec parking space.

**Pull request merged**